src/main/java/org/myrobotlab/service/InMoovHead.java

```diff
@@ -438,22 +373,49 @@ public void setSpeed(Double headXSpeed, Double headYSpeed, Double eyeXSpeed, Dou
   public void startService() {
     super.startService();
 
+    // part of the "start" life-cycle - these services "check" on late binding
+    // meaning if "stuff is already setup - we don't mess with it", otherwise
+    // configure with defaults
+
     if (controller == null) {
       controller = (ServoController) startPeer("arduino");
     }
 
-    jaw.startService();
-    eyeX.startService();
-    eyeY.startService();
-    rothead.startService();
-    neck.startService();
-    rollNeck.startService();
-  }
+    if (jaw == null) {

```

Comment:
all this hard coded configuration stuff should be moved elsewhere form startService. from a lifecycle perspective i would see it as 3 separate phases.   1. create service .   2. configure service .. 3. start service.    
having this code here forces 3 and 2 to occur no matter what.

Reply:
agree in general
Create service means creating peers or not ?  I prefer not.
If you taking the trouble to create a service and not using the default peers or config. Then you should be handed a thin skeleton of the composite service.  Allowing you to set and modify to your delight WITHOUT having to release services that were auto-magically created for you.

To not burden the consumer of releasing pre-existing services, i thought late binding to peers would be a good thing (ie peers created and configured at startService if not already set by previous customizations)

Since peers are not created automagically untill startService - these would all be NPEs

I guess you're arguing to create and configure it all, then make the user tear it down & release it - if they want to do customizations ..

I'm on the fence, but creation does mean fully created - which means I'm prolly leaning towards your side.